### PR TITLE
Bug 2084331: pkg/manifest/manifest.go: Fix checkResourceEnablement

### PR DESCRIPTION
**Pull request by jottofar**
so a resource is filtered out if any of the capabilities are disabled.

### CODE REVIEW

File: pkg/manifest/manifest_test.go
Comment (wking) [inline pkg/manifest/manifest_test.go:724]: nit: Why "works"?  Although I guess you have `enabled capability works` in the previous entry, so maybe the pattern is just to end all test-case names with "works".

File: pkg/manifest/manifest_test.go
Comment (wking) [inline pkg/manifest/manifest_test.go:725]: nit: since #1343's pivot to pointer, we can leave unset all the properties that aren't the one whose filter we're exercising.

File: pkg/manifest/manifest.go
Comment (wking) [inline pkg/manifest/manifest.go:175]: nit: `is*Cap` are only used within the loop, so you can shift the whole `var ...` or `isKnownCap := false` or whatever into the loop and drop the `var` declarations here.

**Review by wking — APPROVED** [review]
/bugzilla refresh
/lgtm
/hold

Left a few comments inline, but nothing that I think needs to block merging.  Ping me for a fresh LGTM if you decide to adjust anything, or pull the hold if you're fine as things stand.

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**
